tests/tape/tapes/test_qnode.py

```diff
@@ -915,3 +915,21 @@ def circuit(x):
         # test differentiability. The circuit will assume an RZ gate
         grad = qml.grad(circuit)(-0.5)
         np.testing.assert_allclose(grad, 0, atol=tol, rtol=0)
+
+
+class TestShots:

```

Comment:
nit: You don't need to add a class when using pytest, you can just use a standalone test function (ignore if you want to stay consistent with rest of file).

Reply:
See [The Walrus](https://github.com/XanaduAI/thewalrus/blob/master/thewalrus/tests/test_quantum.py) test suite for example.

I think pytest recommends classes only when needed for organizational uses.

Reply:
Oh, but this one will grow in the following changes which also need tests!